test/error.jl

```diff
@@ -0,0 +1,72 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+
+@test map(typeof, map(@catch(i->[1,2,3][i]), 1:6)) ==
+      [Int, Int, Int, BoundsError, BoundsError, BoundsError]
+
+@test typeof(@catch(open)("/no/file/with/this/name")) == SystemError
+
+
+let
+
```

Comment:
These excess empty lines at the start and end of `function`, `let`, etc blocks are really visually distracting and serve no purpose. Please conform to the prevailing style used everywhere else in the code base that does not do this.